### DISCUSSION TITLE
How can I get value from json whose key has dot?

### DISCUSSION THREAD

**Discussion by boson234** [trinodb/trino#20249] [Q&A] [answered]
I have a json string，it looks like  json '{"id.a": 1, "value": "xxx"}' ，
I tried to get the value of "id.a", and use "json_extract_scalar"
SELECT json_extract_scalar(json '{"id.a": 1, "value": "xxx"}', '$.id.a');
but the result is null.
I should treat "id.a" as whole,but I don't know how

**Comment by yuokada** [comment]
Maybe, you have to escape like below. 
```sql
SELECT json_extract_scalar(json '{"id.a": 1, "value": "xxx"}', '$."id.a"');
```

I don't confirm the above query actually works. But, it will work.

**Reply by boson234** [reply]
![image](https://github.com/trinodb/trino/assets/34983864/e34030eb-97f0-4720-84d6-c630770f8919)
I tried this, but failed. 
and the version is 435

**Reply by hashhar** [reply]
Use square brackets. Like so: `$["id.a"]`

**Comment by hashhar** [comment, marked as answer]
You have to escape special characters in JSON path expressions using square brackets and quotes:

```sql
SELECT json_extract_scalar(json '{"id.a": 1, "value": "xxx"}', '$["id.a"]');
```

**Reply by boson234** [reply]
it doesn't work too.
![image](https://github.com/trinodb/trino/assets/34983864/fdb31560-8515-465b-8031-15fb836b16e8)

**Reply by hashhar** [reply]
sorry, forgot to remove the `.`. Updated the answer.

`.` is not needed in the bracket notation.

**Reply by boson234** [reply]
yes，it works。thank you！
